modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/test.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+// Your editor should now give you type help for these below
+
+Liferay.Address.getCountries();
+Liferay.Address.getRegions();
+Liferay.Util.fetch();
+Liferay.Util.formatStorage();
+Liferay.Util.formatXML();
+
+// Note that this definition doesn't exist, so TS complains about it.
+
+Liferay.SomeNamespace.foo();
```

Comment:
@bryceosterhaus so the types will only work in TS files? That makes sense. But, does that mean that the types will only work in future code, not in existing code (unless we start changing JS to TS files)?

Reply:
@kresimir-coko yeah the types will only work in TS files, javascript doesn't have a type checking system so we wouldn't expect it to work in js files.

Once we get the Liferay object properly typed, this will allow us to more easily upgrade other modules to TS that are using `Liferay`